glib-macros/src/utils.rs

```diff
@@ -168,34 +168,27 @@ const KNOWN_GLIB_EXPORTS: [&str; 10] = [
 pub fn crate_ident_new() -> TokenStream {
     use proc_macro_crate::FoundCrate;
 
-    let crate_path = match crate_name("glib") {
+    match crate_name("glib") {
         Ok(FoundCrate::Name(name)) => Some(name),
         Ok(FoundCrate::Itself) => Some("glib".to_string()),
         Err(_) => None,
     }
     .map(|s| {
         let glib = Ident::new(&s, Span::call_site());
         quote!(#glib)
-    });
-
-    let crate_path = crate_path.or_else(|| {
-        KNOWN_GLIB_EXPORTS.iter().find_map(|c| {
-            if let Ok(f) = crate_name(c) {
-                let crate_name = match f {
-                    FoundCrate::Name(name) => name,
```

Comment:
what's the reason for dropping this?

Reply:
See the PR description: it stopped at first known crate found in Cargo.toml, even if optional.

Reply:
That would be an issue to report to  proc-macro-crate though, no?

Reply:
Might be yes. I'm away though, so can't take care of it this weekend.

Reply:
I'm not sure why it fails in this particular case though.

Reply:
It fails because gtk4 re-exports glib and we expect people to use the re-exported glib without having to re-add glib as a dependency. So the applications/examples don't directly on glib at all but indirectly through gtk4/gtk3. I have opened https://github.com/gtk-rs/gtk-rs-core/issues/127 so it's not get forgotten

Reply:
Yes, that's the use case I addressed too (in gst-plugins-rs/utils).

Reply:
Ah I see, I think the issue might come from the Downgrade/Upgrade traits.